src/main/java/hypersquare/hypersquare/dev/code/player/action/PlayerLaunchProjectile.java

```diff
@@ -0,0 +1,160 @@
+package hypersquare.hypersquare.dev.code.player.action;
+
+import hypersquare.hypersquare.dev.BarrelParameter;
+import hypersquare.hypersquare.dev.BarrelTag;
+import hypersquare.hypersquare.dev.action.Action;
+import hypersquare.hypersquare.dev.codefile.data.CodeActionData;
+import hypersquare.hypersquare.dev.value.type.DecimalNumber;
+import hypersquare.hypersquare.item.action.ActionItem;
+import hypersquare.hypersquare.item.action.ActionMenuItem;
+import hypersquare.hypersquare.item.action.player.PlayerActionItems;
+import hypersquare.hypersquare.item.value.DisplayValue;
+import hypersquare.hypersquare.menu.barrel.BarrelMenu;
+import hypersquare.hypersquare.play.CodeSelection;
+import hypersquare.hypersquare.play.execution.ExecutionContext;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.attribute.Attribute;
+import org.bukkit.attribute.AttributeModifier;
+import org.bukkit.entity.*;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.projectiles.ProjectileSource;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+public class PlayerLaunchProjectile implements Action {
+    @Override
+    public void execute(@NotNull ExecutionContext ctx, @NotNull CodeSelection targetSel) {
+        ProjectileType projectileType = ctx.getTag("projectile", ProjectileType::valueOf);
+        float projectileSpeed = ctx.args().getOr("speed", new DecimalNumber(1, 0)).toFloat();
+        float projectileInaccuracy = ctx.args().getOr("inaccuracy", new DecimalNumber(0, 0)).toFloat();
+        for (Player p : targetSel.players()) {
+            Location launchPoint = ctx.args().getOr("location", p.getEyeLocation().add(p.getEyeLocation().getDirection().multiply(1.05)));
+            launchPoint.setPitch((float) ((Math.random()*2-1) * projectileInaccuracy + launchPoint.getPitch()));
+            launchPoint.setYaw((float) ((Math.random()*2-1) * projectileInaccuracy + launchPoint.getYaw()));
```

Comment:
Inaccuracy should be applied to the direction of the projectile, not the launch point.